File: happyShopping_sellergoods_interface/src/main/java/com/happyShopping/sellergoods/service/ItemCatService.java
package com.happyShopping.sellergoods.service;
import java.util.List;

import com.happyShopping.model.ItemCat;
import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemCatService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<ItemCat> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(ItemCat itemCat);
	
	
	/**
	 * 修改
	 */
	public void update(ItemCat itemCat);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public ItemCat findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(ItemCat itemCat, int pageNum, int pageSize);

	/**
	 * 通过上级分类id得到商品分类列表
	 * @param id
	 * @return
	 */
	public List<ItemCat> findByParentId(Long id);
}

File: happyShopping_shop_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../itemCat/findAll.action');
	};
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../itemCat/findPage.action?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../itemCat/findOne.action?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../itemCat/add.action',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../itemCat/update.action',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../itemCat/delete.action?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../itemCat/search.action?page='+page+"&rows="+rows, searchEntity);
	};
	// 通过上级分类id得到商品分类列表
	this.findByParentId = function (parentId) {
		return $http.get('../itemCat/findByParentId.action?id='+parentId);
    }
});
